internal/validations/clusterlogforwarder/outputs/splunk_test.go

```diff
@@ -0,0 +1,89 @@
+package outputs
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	loggingv1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/status"
+	. "github.com/openshift/cluster-logging-operator/test/matchers"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var _ = Describe("Validate Splunk:", func() {
+
+	Context("when validating Output Type", func() {
+
+		It("should be valid", func() {
+			splunk := &loggingv1.Splunk{
+				IndexKey: "kubernetes.labels.test_logging",
+			}
+			valid, st := VerifySplunk("splunk", splunk)
+			Expect(valid).To(BeTrue())
+			Expect(map[string]status.Condition{"splunk": st}).To(HaveCondition("Ready", true, "", ""))
+		})
+
+		It("should fail if no IndexKey and ", func() {
+			splunk := &loggingv1.Splunk{
+				IndexKey:  "kubernetes.labels.test_logging",
+				IndexName: "application",
+			}
+			valid, st := VerifySplunk("splunk", splunk)
+			Expect(valid).To(BeFalse())
+			Expect(map[string]status.Condition{"azureMonitor": st}).To(HaveCondition("Ready", false, loggingv1.ReasonInvalid, "output \"splunk\": Only one of indexKey or indexName can be set, not both."))
```

Comment:
@Clee2691 is this consistent with the epic?

Reply:
Yes this is consistent. We only allow one of `IndexKey` or`IndexName` not both.